### PR TITLE
(#459) CMS Embed Featured Content tests 

### LINKED ISSUE #459: CMS: Embed Featured content
## As a [role], I want [feature], so that [benefit]

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Feature: As a cms user I want to be able to embed Featured contents in Article content type

    Scenario: User is adding new test Article content type for embedding it as a Featured Content
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Add content" action button
        And user clicks on "Article" content type
        Then page title is "Create Article"
        When user selects test site section
        And user fills out the following fields
            | fieldLabel               | value                                                                | field_name                     |
            | Pretty URL               | test-article-for-testing                                             | field_pretty_url               |
            | Page Title               | Automated Test Article for Featured Content Test                     | title                          |
            | Browser Title            | Automated Test Article for Featured Content Test - Browser Title     | field_browser_title            |
            | Card Title               | Automated Test Article for Featured Content Test - Card Title        | field_card_title               |
            | Meta Description         | Automated Test Article for Featured Content Test - Meta Description  | field_page_description         |
            | Feature Card Description | Automated Test Article for Featured Content Test - Feature Card Desc | field_feature_card_description |
        And user enters "Article Heading1" as 1 body section heading
        And user fills out 1 "Body" text area with "This article will be used for testing Featured Content"
        And user selects 1 Lead Image from the list of images
        And browser waits
        And user selects 3 Promotional Image from the list of images for featured content article
        And browser waits
        And user remembers the source of selected promo image for further verification in the featured content article
        And user selects "Published" from "Save as" dropdown
        Then user saves the content page


    Scenario: User is adding new test Blog post content type for embedding it as a Featured Content
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Add content" action button
        And user clicks on "Blog Post" content type
        Then page title is "Create Blog Post"
        And user selects "Cancer Currents" from Blog Series dropdown
        And browser waits
        And user fills out the following fields
            | fieldLabel               | value                                                                  | field_name                     |
            | Pretty URL               | blog-post-for-testing                                                  | field_pretty_url               |
            | Page Title               | Automated Test Blog Post for Featured Content Test                     | title                          |
            | Browser Title            | Automated Test Blog Post for Featured Content Test - Browser Title     | field_browser_title            |
            | Card Title               | Automated Test Blog Post for Featured Content Test - Card Title        | field_card_title               |
            | Meta Description         | Automated Test Blog Post for Featured Content Test - Meta Description  | field_page_description         |
            | Feature Card Description | Automated Test Blog Post for Featured Content Test - Feature Card Desc | field_feature_card_description |
            | Author                   | Automated Test Blog Post for Featured Content Test Author              | field_author                   |
        And user fills out "Body" text area with "This blog post will be used for testing featured content"
        And user selects 1 Lead Image from the list of images
        And browser waits
        And user selects 3 Promotional Image from the list of images for blog post
        And browser waits
        And user remembers the source of selected promo image for further verification in the featured content blog post
        And user selects "Published" from "Save as" dropdown
        Then user saves the content page


    Scenario: User is adding new Article content type and embedding featured contents created above
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Add content" action button
        And user clicks on "Article" content type
        Then page title is "Create Article"
        When user selects test site section
        And user fills out the following fields
            | fieldLabel               | value                                                               | field_name                     |
            | Pretty URL               | test-article-featured-content                                       | field_pretty_url               |
            | Page Title               | Automated Test Article to test Featured Content                     | title                          |
            | Browser Title            | Automated Test Article to test Featured Content - Browser Title     | field_browser_title            |
            | Card Title               | Automated Test Article to test Featured Content - Card Title        | field_card_title               |
            | Meta Description         | Automated Test Article to test Featured Content - Meta Description  | field_page_description         |
            | Feature Card Description | Automated Test Article to test Featured Content - Feature Card Desc | field_feature_card_description |
        And user enters "Article List Description" into "List Description" text field
        And user enters "Article Intro Text" as intro text
        And user enters "Article Heading1" as 1 body section heading
        And user fills out 1 "Body" text area with "Embedding Featured Content: Article Feature Card: Image None"
        And user clicks the "Featured Content" button 1 in the WYSIWYG editor
        And user enters "Automated Test Article for Featured Content Test" into content title search box and clicks "Apply"
        And user selects "Automated Test Article for Featured Content Test" item from the content list
        And browser waits
        And user clicks on "Select content" button to select content
        And browser waits
        And "Display as" dropdown has the following options
            | options                |
            | Feature Card: Image    |
            | Feature Card: No Image |
        And "Display as" dropdown displays "Feature Card: Image"
        And "Align" label has the following options
            | options |
            | None    |
            | Left    |
            | Center  |
            | Right   |
        And the radio button "None" is selected by default under "Align"
        And user clicks on "Embed" button to select featured content
        And browser waits
        And user clicks on "Add Body Section" to add a body section
        And user enters "Article Heading2" as 2 body section heading
        And user fills out 2 "Body" text area with "Embedding Featured Content: Article Feature Card: No Image Left"
        And user clicks the "Featured Content" button 2 in the WYSIWYG editor
        And user enters "Automated Test Article for Featured Content Test" into content title search box and clicks "Apply"
        And user selects "Automated Test Article for Featured Content Test" item from the content list
        And browser waits
        And user clicks on "Select content" button to select content
        And user select "Feature Card: No Image" from the "Display as" dropdown
        And browser waits
        And user selects "left" radio button under "Align"
        And user clicks on "Embed" button to select featured content
        And browser waits
        And user clicks on "Add Body Section" to add a body section
        And user enters "Article Heading3" as 3 body section heading
        And user fills out 3 "Body" text area with "Embedding Featured Content: Blog Post Feature Card: Image Center"
        And user clicks the "Featured Content" button 3 in the WYSIWYG editor
        And user enters "Automated Test Blog Post for Featured Content Test" into content title search box and clicks "Apply"
        And user selects "Automated Test Blog Post for Featured Content Test" item from the content list
        And browser waits
        And user clicks on "Select content" button to select content
        And user select "Feature Card: Image" from the "Display as" dropdown
        And browser waits
        And user selects "center" radio button under "Align"
        And user clicks on "Embed" button to select featured content
        And browser waits
        And user clicks on "Add Body Section" to add a body section
        And user enters "Article Heading4" as 4 body section heading
        And user fills out 4 "Body" text area with "Embedding Featured Content: Blog Post Feature Card: No Image Right"
        And user clicks the "Featured Content" button 4 in the WYSIWYG editor
        And user enters "Automated Test Blog Post for Featured Content Test" into content title search box and clicks "Apply"
        And user selects "Automated Test Blog Post for Featured Content Test" item from the content list
        And browser waits
        And user clicks on "Select content" button to select content
        And user select "Feature Card: No Image" from the "Display as" dropdown
        And browser waits
        And user selects "right" radio button under "Align"
        And user clicks on "Embed" button to select featured content
        And browser waits
        And user selects 1 Lead Image from the list of images
        And browser waits
        And user selects 3 Promotional Image from the list of images on the article
        And browser waits
        And user selects "Published" from "Save as" dropdown
        Then user saves the content page


    Scenario: Verify embedded Featured Contents at the front end in the English Article
        Given user is navigating to the front end site with path site section plus "test-article-featured-content"
        And browser waits
        And screen breakpoint is set to "desktop"
        And page title is "Automated Test Article to test Featured Content"
        And the intro text reads "Article Intro Text"
        And 1 section heading reads "Article Heading1"
        And 1 description reads "Embedding Featured Content: Article Feature Card: Image None"
        And 1 feature card displays the following
            | title                                                         | link                                         | description                                                          |
            | Automated Test Article for Featured Content Test - Card Title | {TEST_SITE_SECTION}/test-article-for-testing | Automated Test Article for Featured Content Test - Feature Card Desc |
        And the promo image is matching the earlier selected promo image in the article to be used for embeddding
        And 2 section heading reads "Article Heading2"
        And 2 description reads "Embedding Featured Content: Article Feature Card: No Image Left"
        And 2 feature card displays the following
            | title                                                         | link                                         | description                                                          |
            | Automated Test Article for Featured Content Test - Card Title | {TEST_SITE_SECTION}/test-article-for-testing | Automated Test Article for Featured Content Test - Feature Card Desc |
        And 2 feature card does not display any image
        And 3 section heading reads "Article Heading3"
        And 3 description reads "Embedding Featured Content: Blog Post Feature Card: Image Center"
        And 3 feature card displays the following
            | title                                                           | link                                                          | description                                                            |
            | Automated Test Blog Post for Featured Content Test - Card Title | news-events/cancer-currents-blog/{YEAR}/blog-post-for-testing | Automated Test Blog Post for Featured Content Test - Feature Card Desc |
        And the promo image is matching the earlier selected promo image in the blog post to be used for embeddding
        And 4 section heading reads "Article Heading4"
        And 4 description reads "Embedding Featured Content: Blog Post Feature Card: No Image Right"
        And 4 feature card displays the following
            | title                                                           | link                                                          | description                                                            |
            | Automated Test Blog Post for Featured Content Test - Card Title | news-events/cancer-currents-blog/{YEAR}/blog-post-for-testing | Automated Test Blog Post for Featured Content Test - Feature Card Desc |
        And 4 feature card does not display any image


    Scenario: Clean up
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user selects a checkbox next to title with url "test-article-for-testing" from the list of content
        And user selects a checkbox next to title with url "blog-post-for-testing" under "/news-events/cancer-currents-blog" from the list of content
        And user selects a checkbox next to title with url "test-article-featured-content" from the list of content
        And user clicks on "Apply to selected items" content action button
        Then page title is "Are you sure you want to delete these content items?"
        When user clicks on "Delete" button
        Then the confirmation text "Deleted 3 content items" appears on a screen
        And the content item with url "test-article-for-testing" does not exist in the list of content
        And the content item with url "blog-post-for-testing" does not exist in the list of content
        And the content item with url "test-article-featured-content" does not exist in the list of content
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Closes #459 

Added feature file and index.js

### CODE REVIEW

**Comment by alinai** [comment]
Test run on test-ac:
image

Test run on dev-ac:
image

**Comment by alinai** [comment]
@KateMashkinaNIH , please review.

**Comment by alinai** [comment]
Test run on dev-ac after moving the test to WYSIWYG folder:

image

File: cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        And user clicks on "Blog Post" content type
+        Then page title is "Create Blog Post"
+        And user selects "Cancer Currents" from Blog Series dropdown
+        And browser waits
+        And user fills out the following fields
+            | fieldLabel               | value                                                                  | field_name                     |
+            | Pretty URL               | blog-post-for-testing                                                  | field_pretty_url               |
+            | Page Title               | Automated Test Blog Post for Featured Content Test                     | title                          |
+            | Browser Title            | Automated Test Blog Post for Featured Content Test - Browser Title     | field_browser_title            |
+            | Card Title               | Automated Test Blog Post for Featured Content Test - Card Title        | field_card_title               |
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature:50]: I suggest to remove card title form here so we can verify Browser title shows as card title if its empty on the feature cards

File: cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        Then page title is "Create Article"
+        When user selects test site section
+        And user fills out the following fields
+            | fieldLabel               | value                                                                | field_name                     |
+            | Pretty URL               | test-article-for-testing                                             | field_pretty_url               |
+            | Page Title               | Automated Test Article for Featured Content Test                     | title                          |
+            | Browser Title            | Automated Test Article for Featured Content Test - Browser Title     | field_browser_title            |
+            | Card Title               | Automated Test Article for Featured Content Test - Card Title        | field_card_title               |
+            | Meta Description         | Automated Test Article for Featured Content Test - Meta Description  | field_page_description         |
+            | Feature Card Description | Automated Test Article for Featured Content Test - Feature Card Desc | field_feature_card_description |
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature:21]: remove feature card description from here - blog posts will have description and article wont ( 2 different use cases)

File: cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            | Page Title               | Automated Test Blog Post for Featured Content Test                     | title                          |
+            | Browser Title            | Automated Test Blog Post for Featured Content Test - Browser Title     | field_browser_title            |
+            | Card Title               | Automated Test Blog Post for Featured Content Test - Card Title        | field_card_title               |
+            | Meta Description         | Automated Test Blog Post for Featured Content Test - Meta Description  | field_page_description         |
+            | Feature Card Description | Automated Test Blog Post for Featured Content Test - Feature Card Desc | field_feature_card_description |
+            | Author                   | Automated Test Blog Post for Featured Content Test Author              | field_author                   |
+        And user fills out "Body" text area with "This blog post will be used for testing featured content"
+        And user selects 1 Lead Image from the list of images
+        And browser waits
+        And user selects 3 Promotional Image from the list of images for blog post
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature:57]: also remove promo image from blog - lead image should display instead on FF if you dont provide a promo override
article will have a promo override and blog should display lead ( again 2 use cases)

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

File: cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature (hunk 1 above)
Comment (alinai) [reply cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature:50]: Done

File: cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature (hunk 2 above)
Comment (alinai) [reply cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature:21]: done

File: cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature (hunk 3 above)
Comment (alinai) [reply cypress/e2e/CMS/WYSIWYG/$EmbedFeaturedContent.feature:57]: done

**Comment by alinai** [comment]
Test run on test-ac after updating per review:

image


@KateMashkinaNIH , please review!

**Comment by alinai** [comment]
Test run on test-acsf:
image

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**